lib/get-specs-from-issue.js

```diff
@@ -2,38 +2,81 @@ module.exports = getSpecsFromIssue;
 
 function getSpecsFromIssue(context, issueNumber) {
   console.log('started getting specs out of issue');
-  let issue = context.github.issues;
-  let specs = [];
+  const repo = context.repo();
 
   return Promise.resolve()
-  .then(() => Promise.all([
-    context.github.issues.get(...repo, issueNumber),
-    context.github.issues.listComments(...repo, issueNumber)
-  ]))
-  .then(data => {
-    const { body, comments } = data;
-    const allCommentsBody = comments
-      .map((comment) => comment.body)
-      .push(body);
-
-    pushSpecs(specs, allCommentsBody);
-
-    return specs;
-  });
+    .then(() => Promise.all([
+      context.github.issues.get(...repo, issueNumber),
+      context.github.issues.listComments(...repo, issueNumber)
+    ]))
+    .then(data => {
+      const { issueBody, issueComments } = data;
+
+      return issueComments
+        .map(comment => comment.body)
+        .push(issueBody)
+        .join('\n');
+    })
+    .then(issueCompleteText => getSpecsArray(issueCompleteText))
+    .then(specsArray => specNamesFromMarkdownList(specsArray));
 }
 
-function pushSpecs(specs, body) {
-  body.toLowerCase();
-  let posiotionOfSpecsList = body.indexOf('## specs');
-  body = body.slice(posiotionOfSpecsList);
-  let endPosiotionOfSpecsList = body.indexOf('##');
-  body = body.slice(0, endPosiotionOfSpecsList);
-
-  let bullets = body.split('-');
-  bullets.forEach(bullet => {
-    bullet.trim();
-    if (bullet.startsWith('it')) {
-      specs.push(bullet);
+function getSpecsArray(completeText) {
+  const specsBlockStartRegex = /^\s*?(#+)\s*?specs/gi
+  const specsArray = [];
```

Comment:
you can remove the `Array` suffix from this parameter and use something more informative like `specNames`

please see [The art of naming variables](https://hackernoon.com/the-art-of-naming-variables-52f44de00aad)